src/ConnectedMode/SlCoreConnectionAdapter.cs

```diff
@@ -98,6 +103,39 @@ public Task<AdapterResponseWithData<List<OrganizationDisplay>>> GetOrganizations
         });
     }
 
+    public Task<AdapterResponseWithData<ServerProject>> GetServerProjectByKeyAsync(ServerConnection serverConnection, ConnectionInfo connectionInfo, string serverProjectKey)
+    {
+        var failedResponse = new AdapterResponseWithData<ServerProject>(false, null);
+        
+        return threadHandling.RunOnBackgroundThread(async () =>
+        {
+            if (!TryGetConnectionConfigurationSlCoreService(out var connectionConfigurationSlCoreService))
+            {
+                return failedResponse;
+            }
+
+            try
+            {
+                var credentials = (BasicAuthCredentials) serverConnection.Credentials;
+                if (credentials == null)
+                {
+                    return failedResponse;
+                }
+                var credentialsSlCoreFormat = credentials.Password?.Length > 0
+                    ? Either<TokenDto, UsernamePasswordDto>.CreateRight(new UsernamePasswordDto(credentials.UserName, credentials.Password.ToString()))
+                    : Either<TokenDto, UsernamePasswordDto>.CreateLeft(new TokenDto(credentials.UserName));
+                var transientConnection = GetTransientConnectionDto(connectionInfo, credentialsSlCoreFormat);
+                var response = await connectionConfigurationSlCoreService.GetProjectNamesByKeyAsync(new GetProjectNamesByKeyParams(transientConnection, [serverProjectKey]));
+                return new AdapterResponseWithData<ServerProject>(true, new ServerProject(serverProjectKey, response.projectNamesByKey[serverProjectKey]));
```

Comment:
When I just bind a project and the SlCore can not find the ProjectKey in the server, it returns a Dictionary where the projectKey is present, but the name is null. In that case, should the response be true or false? It is actually not found...